src/language/records.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+_Records expressions_ are comma-delimited lists of named or positional fields,
+enclosed in parentheses:
+
+```dart
+// Record expression:
+var record = ('first', a: 2, b: true, 'last');
+```
+
+_Record type annotations_ are comma-delimited lists of types enclosed in parentheses.
+The `(int, int)` statements in the following example are record type annotations
+used as a return type and as parameter types:
+
+```dart
+(int, int) swap((int, int) record) {
+  var (a, b) = record;
+  return (b, a);
+}
+```
+
+Named fields inside _record type annotations_ mirror how [named parameters and arguments][]
+work in functions. They go inside a curly brace-delimited
+section of type-and-name pairs, after all positional fields:
+
+```dart
+// Record type annotation in a variable declaration:
+(String, String?, {int a, bool b}) record;
+```
+
+Naming fields in record type annotations is optional for documentation purposes.
+Once you create a record from a type annotation with named fields using a record
+expression, the names become part of the [record's type definition](#record-types),
+or its _shape_:
+
+```dart 
+  ({int x, int y}) someRecord = (x: 1, y: 2);
+// ^^^^^^^^^^^^^^^              ^^^^^^^^^^^^
+// type annotation              expression
+```
+
+They can be ommitted from the creation expression, but not replaced
+with other names.
+
+```dart
+// Type annotation:
+({int x, int y}) someRecord;
+// Record expression creating record from type annotation, omitting names:
+someRecord = (1, 2);
+// Prints 1, 2:
+print(someRecord.x, someRecord.y);
+```
+
+For more information and examples, check out [Record types](#record-types) and
+[Record equality](#record-equality).
```

Comment:
@munificent this was the only significant change from the last commit (lines 34-67), could you review it one more time? There were a couple comments on the confusing wording so I tried to elaborate

Reply:
It's getting there, but still not quite accurate. How about:

---

Fields in record expressions and type annotations mirror how [parameters and arguments][] work in functions. Positional fields go directly inside the parentheses:

```dart
// Record type annotation in a variable declaration:
(String, int) record;

// Initialize it with a record expression:
record = ('A string', 123);
```

In a record type annotation, named fields go inside a curly brace-delimited
section of type-and-name pairs, after all positional fields. In a record
expression, the names go before each field value with a colon after:

```dart
// Record type annotation in a variable declaration:
({int a, bool b}) record;

// Initialize it with a record expression:
record = (a: 123, b: true);
```

The names of named fields in a record type are part of thee [record's type
definition](#record-types), or its _shape_. Two records with named fields with
different names have different types:

```dart
({int a, int b}) recordAB = (a: 1, b: 2);
({int x, int y}) recordXY = (x: 3, y: 4);

recordAB = recordXY; // Compile error! These records don't have the same type.
```

In a record type annotation, you can also name the *positional* fields, but
these names are purely for documentation and don't affect the record's type:

```dart
(int a, int b) recordAB = (1, 2);
(int x, int y) recordXY = (3, 4);

recordAB = recordXY; // OK.
```

This is similar to how positional parameters in a function declaration or function typedef can have names but those names don't affect the signature of the function.

For more information and examples, check out [Record types](#record-types) and
[Record equality](#record-equality).

Reply:
Thanks, I'll open another PR for this!